typescript/README.md

```diff
@@ -1 +1,178 @@
 # TypeScript Best Practices
+
+1. **Consistent Coding Style**:
+   - **Explanation**: Use a linter like ESLint to maintain a consistent coding style.
+      - @typescript-eslint/parser
+      - @typescript-eslint/eslint-plugin
+      - eslint-config-airbnb-typescript
+
+   Configure ESLint: Create or modify the `.eslintrc.json` file at the root of your project with the following configuration:
+
+   ```json
+    {
+        "extends": [
+            "airbnb-typescript"
+        ],
+        "parser": "@typescript-eslint/parser",
+        "parserOptions": {
+            "project": "./tsconfig.json"
+        },
+        "plugins": [
+            "@typescript-eslint"
+        ],
+        "rules": {
+            "semi": ["error", "always"],
+            "quotes": ["error", "single"],
+            "no-unused-vars": "error",
+            "eqeqeq": ["error", "always"],
+            "curly": ["error", "all"],
+            "no-console": "warn",
+            "consistent-return": "error",
+            "prefer-const": "error"
+        }
+    }
+    ```
+
+   This configuration allows you to benefit from Airbnb's best practices while integrating TypeScript specifics.
+
+2. **Type Annotations**:
+   - **Explanation**: Provide type annotations for function arguments and return values to ensure type safety.
+   ```typescript
+   function greet(name: string): string {
+     return `Hello, ${name}`;
+   }
+   ```
+
+3. **Interfaces**:
+   - **Explanation**: Prefer interfaces over type aliases for defining object shapes.
+   ```typescript
+   interface User {
+     name: string;
+     age: number;
+   }
+   ```
+
+4. **Avoid `any`**:
+   - **Explanation**: Using `any` defeats the purpose of TypeScript's type safety.
+   ```typescript
+   let data: any; // Avoid this
+   let userData: User; // Prefer this
+   ```
+
+5. **Strict Null Checks**:
+   - **Explanation**: Enable strict null checks to prevent null or undefined errors.
+   ```typescript
+   let nullableString: string | null = null;
+   ```
+
+6. **Use `readonly` for Immutability**:
+   - **Explanation**: Use `readonly` for properties that should not be modified after initialization.
+    ```typescript
+    interface User {
+      readonly id: number;
+      name: string;
+      age: number;
+    }
+    ```
+
+7. **Prefer `const` over `let`**:
```

Comment:
this is more related to JS than a specific TS feature, it's common for both JS and TS.
We can have as prerequisite at the begin for example: Adhere to JS best practices => link to JS gems.